docs/xml.md

```diff
@@ -1,6 +1,681 @@
+
+## External
+
+This element is a legacy feature and currently does nothing. Roblox Studio encodes two of these elements when writing files, but their presence is OPTIONAL. When present, they MUST be under the `roblox` element.
+
+There are no attributes required for this element.
+
+The contents of this element represent an unknown purpose.
+
+## Item
+
+This element describes one `Instance` value. All `Item` elements MUST be under either the `roblox` element or other `Item` elements.
+
+The following attributes are REQUIRED for this element:
+
+| Name       | Contents                                                              |
+|:-----------|:----------------------------------------------------------------------|
+| `class`    | The class of the `Instance` this element represents.                  |
+| `referent` | A unique string used to reference this element elsewhere in the file. |
+
+The value of `referent` MUST be unique for the file. The value of `referent` MUST NOT be `null`. Roblox utilizes the value `null` when serializing [`Ref`](#ref) properties with no value, so it should be considered a reserved value.
+
+Roblox generates referents by prefixing a UUID with `RBX`, but this is not a requirement.
+
+It is suggested that every file have at least `Item` value as otherwise there is no purpose to the file.
+
+## Properties
+
+This element contains all properties for a given `Instance`. There MUST be one `Properties` element per `Item`. All `Properties` elements MUST be under an `Item` element.
+
+There are no attributes required for this element.
+
+Every child of this element is a [Type Element](#type-elements) and represents exactly one property of an `Instance`.
+
+## SharedStrings
+
+This element acts as a repository for [`SharedString` definitions][SharedString-def]. There MAY be zero or one `SharedStrings` elements per file. When present, the `SharedStrings` element MUST be under the `roblox` element.
+
+There are no attributes required for this element.
+
+## SharedString
+[SharedString-def]: #sharedstring
+
+This element defines a single `SharedString` value for reference by [Type Elements](#type-elements). There MAY be zero or more `SharedString` elements per file. `SharedString` elements MUST be under the `SharedStrings` element.
+
+This element shares a name with a type element. That element is documented [here][SharedString-use].
+
+The following attributes are REQUIRED for this element:
+
+| Name  | Contents                                                               |
+|:------|:-----------------------------------------------------------------------|
+| `md5` | A unique identifier for this element, for reference by a type element. |
+
+The value of `md5` MUST be unique across all `SharedString` definitions. Despite the name, the value does not have to be the MD5 hash of the `SharedString` contents.
```

Comment:
They don't have to be the MD5 hash of the SharedString contents, but does it need to be MD5 at all? (I suspect no?)

Reply:
The attribute has be named `md5`, if that's what you mean.